src/upgrade.rs

```diff
@@ -316,27 +316,39 @@ mod sealed {
         fn on_upgrade(self) -> OnUpgrade;
     }
 
+    /*
```

Comment:
We should be able to keep this implementation (in fact, removing it is a "breaking change", as you noticed by having to fix up the examples).

Reply:
@seanmonstar Yes, I don't intend to remove this in the end. This is just a temporary kludge in order to solve the cases I could understand how to make compile, since I'm not sure how to work around the immutability. Before finalizing the PR, these cases should definitely be fixed as well.

Are you able to provide guidance on how to implement `CanUpgrade` for `http::Request<crate::Body>` and `http::Response<crate::Body>` though?

Reply:
It'd be the same as for the `&mut _` version, since taking an argument by-value still means you can mutate them (thus calling `extensions_mut().remove::<Thingy>()`.

Reply:
@seanmonstar Thanks for the tip - I tried that strategy before and couldn't hack it, but will try again.

Reply:
@seanmonstar I think it actually worked! Not sure why I struggled so much with it before.